### PR TITLE
Sql DB project dashboard

### PR DESCRIPTION
**Pull request by SakshiS-harma**
Merging prior PRs on dashboard to main(https://github.com/microsoft/azuredatastudio/pull/14160 and https://github.com/microsoft/azuredatastudio/pull/14813).
These PRs added a dashboard for Sql DB projects in dataworkspace extension.
image

### CODE REVIEW

**Comment by kisantia** [comment]
there's a merge conflict and a couple tests failing that need to be fixed.

**Comment by SakshiS-harma** [comment]
> there's a merge conflict and a couple tests failing that need to be fixed.

Fixed the tests and synced to latest to clear out the merge conflict.

**Comment by udeeshagautam** [comment]
Looks good to me, there are no other changes in addition to the signed off PRs mergining - yeah?

**Comment by SakshiS-harma** [comment]
> Looks good to me, there are no other changes in addition to the signed off PRs mergining - yeah?

Apart from the signed off PRs, there are a couple of tests that I fixed (which were missed in the last PR).

**Review by kisantia — APPROVED** [review]
Doesn't have to be in this PR, but can you add some tests for the new dashboard code? This is a lot of new code, so we shouldn't let our code coverage numbers go down :)

**Review by alanrenmsft — APPROVED** [review]

File: extensions/data-workspace/src/dialogs/projectDashboard.ts [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+
+	public async showDashboard(): Promise<void> {
+		const project = this._treeItem.element.project;
+
+		if (!(await fileExist(project.projectFilePath)) || project.projectFileName === null) {
+			throw new Error(constants.fileDoesNotExist(project.projectFilePath));
+		}
+
+		if (project.projectFileName === null) {
+			throw new Error(constants.projectNameNull);
```
[the remark this answers is not in the document]
Comment (SakshiS-harma) [reply extensions/data-workspace/src/dialogs/projectDashboard.ts:32]: If there is no projectFileName, then it wouldn't show up the dashboard at all, instead shows the following error:
image

Either no filePath or fileName just means that the project is somehow corrupted and hence nothing to present in dashoard.

**Comment by SakshiS-harma** [comment]
> Doesn't have to be in this PR, but can you add some tests for the new dashboard code? This is a lot of new code, so we shouldn't let our code coverage numbers go down :)

Sure, I will add that in a separate PR.

**Pull request merged**